spyder/widgets/ipythonconsole/shell.py

```diff
@@ -139,18 +139,20 @@ def short_banner(self):
     def clear_console(self):
         self.execute("%clear")
 
-    def reset_namespace(self):
-        """Resets the namespace by removing all names defined by the user"""
-
-        reply = QMessageBox.question(
-            self,
-            _("Reset IPython namespace"),
-            _("All user-defined variables will be removed."
-            "<br>Are you sure you want to reset the namespace?"),
-            QMessageBox.Yes | QMessageBox.No,
+    def reset_namespace(self, predefined=False):
+        """Reset the namespace by removing all names defined by the user."""
+        reset_str = _("Reset IPython namespace")
+        warn_str = _("All user-defined variables will be removed."
+                     "<br>Are you sure you want to reset the namespace?")
+        if not predefined:
+            reply = QMessageBox.question(self, reset_str,
+                                         warn_str,
+                                         QMessageBox.Yes | QMessageBox.No
             )
 
-        if reply == QMessageBox.Yes:
+            if reply == QMessageBox.Yes:
+                self.execute("%reset -f")
+        else:
             self.execute("%reset -f")
```

Comment:
To avoid saving this line on every execution of `runfile`, please change this to

```python
shellwidget = self.get_current_shellwidget()
if shellwidget is not None:
    shellwidget.silent_execute("%reset -f")
```